src/lib/hooks/useCheckAllowance.ts

```
@@ -0,0 +1,35 @@
+import { useEffect, useState, useRef } from "react";
+import { useApi } from "@/lib/hooks/useApi";
+import { env } from "@/env";
+
+const useCheckAllowance = () => {
+  const [checkingAllowance, setCheckingAllowance] = useState(true);
+  const [isAllowanceSufficient, setAllowanceSufficient] = useState(false);
+
+  const { apiGet } = useApi();
+  const apiGetRef = useRef(apiGet);
+
+  useEffect(() => {
+    const checkAllowance = async () => {
+      setCheckingAllowance(true);
+      try {
+        const isAllowanceSufficient: boolean = await apiGetRef.current(
+          `${env.NEXT_PUBLIC_BACKEND_API_URL}/autoallocator/check_if_allowance_is_sufficient`,
+        );
+        setAllowanceSufficient(isAllowanceSufficient);
+      } catch (error) {
+        console.error(
```

Is logging the error to console and failing silently (no error state handled on UI) by design? Might be good idea to handle the error somehow. Using something like [SWR](https://www.npmjs.com/package/swr) could increase readability and reduce boilerplate, but may not be necessary for now.

---

You can easily add something like
https://sonner.emilkowal.ski

---

Better error handling will be implemented in a separate task.